### PR TITLE
Stop building obsolete COMP services

### PR DESCRIPTION
**Pull request by amaltaro**
Before merging it, a broader discussion is needed. I'm starting a thread in the WebInterfaces HN...

### CODE REVIEW

**Comment by muhammadimranfarooqi** [comment]
+1 to 92ccdc8ef74cae55be14bb8fac28815091cb0422 on slc7_amd64_gcc630.<br>Build log can be found <a href="https://cern.ch/cms-dmwm-builds/6405/test_92ccdc8-20201111-162403.log" target="_blank">here</a>.

File: comp.spec [hunk 1]
```diff
@@ -3,23 +3,26 @@
...
 Requires: popdbweb victorweb confdb exporters exitcodes
-# CMSFOMON
-Requires: happyface sreadiness
 # Common
 Requires: rotatelogs pystack py2-psutil wmcore-devtools
 # Other
 Requires: wmagent-dev condor crabtaskworker t0 couchdb15 crab-devtools
-Requires: PHEDEX-combined-agents PHEDEX-lifecycle lifecycle-das
-Requires: DCAFPilot py2-future compsec wmarchive spacemon-client
+Requires: lifecycle-das
```
Comment (vkuznet) [inline comp.spec]: Alan, this can be removed too

File: comp.spec [hunk 2]
```diff
@@ -3,23 +3,26 @@
...
+Requires: wmarchive
 # Python3
 BuildRequires: wmcorepy3-devtools
+### List of services that are likely no longer needed
+BuildRequires: DMWMMON-datasvc gitweb spacemon-client DCAFPilot compsec
```
Comment (vkuznet) [inline comp.spec]: DCAFPilot can be remove here

**Review by vkuznet — CHANGES_REQUESTED** [review]

File: comp.spec (hunk 1 above)
Comment (amaltaro) [reply comp.spec]: Thanks Valentin. Removing it in my next commit.

File: comp.spec (hunk 2 above)
Comment (amaltaro) [reply comp.spec]: Thanks Valentin. Removing it in my next commit

**Comment by muhammadimranfarooqi** [comment]
+1 to 95d3af2b928e3fd0488cb2ff13b1086621eb84a6 on slc7_amd64_gcc630.<br>Build log can be found <a href="https://cern.ch/cms-dmwm-builds/6405/test_95d3af2-20201111-180503.log" target="_blank">here</a>.

**Comment by muhammadimranfarooqi** [comment]
+1 to f8b88e65389651e3ee91c10876355517303896c5 on slc7_amd64_gcc630.<br>Build log can be found <a href="https://cern.ch/cms-dmwm-builds/6405/test_f8b88e6-20201112-144003.log" target="_blank">here</a>.

**Comment by muhammadimranfarooqi** [comment]
+1 to 09db59996bc8b7b50737c33767e94ca0e385989c on slc7_amd64_gcc630.<br>Build log can be found <a href="https://cern.ch/cms-dmwm-builds/6405/test_09db599-20201113-100503.log" target="_blank">here</a>.

**Comment by muhammadimranfarooqi** [comment]
@amaltaro 

I am getting this error during installation. 

```
++ /data/srv/beHG2012a/sw/common/cmspkg -a slc7_amd64_gcc630 -y install cms+PHEDEX-datasvc+
Reading Package Lists...
error: unknown package: cms+PHEDEX-datasvc+
+++ err=1
+++ note 'ERROR: installation failed with exit code 1.'
+++ echo 'ERROR: installation failed with exit code 1.

```

**Comment by amaltaro** [comment]
Do you have to provide the full RPM name? Or perhaps only "PHEDEX-datasvc"?

**Comment by muhammadimranfarooqi** [comment]
@amaltaro 

I found this from the log. `cms+PHEDEX-datasvc`. Here is complete log

```
++ project=phedex
++ project_version_override=
++ project_config_src=/data/cfg/phedex
++ project_config=/data/srv/current/config/phedex
++ project_auth=/data/srv/beHG2012a/auth/phedex
++ project_state=/data/srv/state/phedex
++ project_logs=/data/srv/logs/phedex
++ variant=default
++ cd /data/srv
++ '[' '!' -d /data/srv/state/phedex ']'
++ cd /data/srv/state/phedex
++ deploy_phedex_deps
++ deploy sw backend
++ local stage=sw
++ shift
++ local pkg version variant
++ case $1 in
++ pkg=backend
++ version=
++ local var=default
+++ echo backend
+++ tr - _
++ local name=backend
+++ echo default
+++ tr - _
++ local varname=default
++ eval local 'did=$backend_default_deployed'
+++ local did=1
++ '[' X1 = X ']'
++ true
+ note 'INFO: deploying phedex - variant: default, version: default'
+ echo 'INFO: deploying phedex - variant: default, version: default'
INFO: deploying phedex - variant: default, version: default
+ echo 'INFO: deploying phedex - variant: default, version: default'
+ eval ':; project='\''phedex'\''; project_version_override='\'''\''; project_config_src='\''/data/cfg/phedex'\''; project_config='\''/data/srv/current/config/phedex'\''; project_auth='\''/data/srv/beHG2012a/auth/phedex'\''; project_state='\''/data/srv/state/phedex'\''; project_logs='\''/data/srv/logs/phedex'\''; variant='\''default'\''; cd $root; [ ! -d $project_state ] || cd $project_state; deploy_phedex_sw'
++ :
++ project=phedex
++ project_version_override=
++ project_config_src=/data/cfg/phedex
++ project_config=/data/srv/current/config/phedex
++ project_auth=/data/srv/beHG2012a/auth/phedex
++ project_state=/data/srv/state/phedex
++ project_logs=/data/srv/logs/phedex
++ variant=default
++ cd /data/srv
++ '[' '!' -d /data/srv/state/phedex ']'
++ cd /data/srv/state/phedex
++ deploy_phedex_sw
++ deploy_pkg comp cms+PHEDEX-datasvc
++ local authfiles=
++ local authfile=
++ local link=
++ '[' 2 -ge 1 ']'
++ case $1 in
++ break
++ '[' 2 = 0 ']'
+++ echo comp
+++ sed -e 's/comp/comp/; n'
++ local repo=comp
+++ echo comp
+++ sed 's/^comp//; s/^\([^.]\)/.\1/; s/^/sw/; n'
++ local label=sw
++ local swarea=/data/srv/beHG2012a/sw
++ local ver=auto
++ local pkg=cms+PHEDEX-datasvc
++ case $ver:$pkg in
++ case $ver:$release in
++ set -e
++ inrepo comp /data/srv/beHG2012a/sw
++ local repo=comp
++ local swarea=/data/srv/beHG2012a/sw
++ '[' '!' -f /data/srv/beHG2012a/sw/.bootstrapped ']'
++ '[' Xauto '!=' Xauto ']'
+++ /data/srv/beHG2012a/sw/common/cmspkg -a slc7_amd64_gcc630 depends cms+comp+HG2012a
+++ fgrep 'Depends: cms+PHEDEX-datasvc+'
+++ awk -F+ '{print $NF}'
++ ver=
++ /data/srv/beHG2012a/sw/common/cmspkg -a slc7_amd64_gcc630 update
Updating cmspkg caches ...
  Remote  caches: 12
  Local   caches: 12
  Caches updated: 0
  Caches deleted: 0
Package cache up-to-date
+++ /data/srv/beHG2012a/sw/common/cmspkg -a slc7_amd64_gcc630 search fake
+++ awk '-F ' '{print $1}'
++ /data/srv/beHG2012a/sw/common/cmspkg -a slc7_amd64_gcc630 -y install external+fakesystem+1.0
Reading Package Lists...
external+fakesystem+1.0 is already the newest version.
0 upgraded, 0 newly installed, 0 removed and 0 not upgraded.
++ /data/srv/beHG2012a/sw/common/cmspkg -a slc7_amd64_gcc630 -y install cms+PHEDEX-datasvc+
Reading Package Lists...
error: unknown package: cms+PHEDEX-datasvc+
+++ err=1
+++ note 'ERROR: installation failed with exit code 1.'
+++ echo 'ERROR: installation failed with exit code 1.'
ERROR: installation failed with exit code 1.
+++ echo 'ERROR: installation failed with exit code 1.'
+++ exit 1
+++ err=1
+++ note 'ERROR: installation failed with exit code 1.'
+++ echo 'ERROR: installation failed with exit code 1.'
ERROR: installation failed with exit code 1.
+++ echo 'ERROR: installation failed with exit code 1.'
+++ exit 1
++ err=1
++ note 'ERROR: installation failed with exit code 1.'
++ echo 'ERROR: installation failed with exit code 1.'
ERROR: installation failed with exit code 1.
++ echo 'ERROR: installation failed with exit code 1.'
++ exit 1
```

**Comment by smuzaffar** [comment]
Do you still need to build `PHEDEX-datasvc` ? if yes then I would suggets to explicitly add it in the comp.spec. Looks like it was indirectly included via one of the servcies which @amaltaro removed. 

If it is not needed then you need to fix your install script to not search for it

**Comment by muhammadimranfarooqi** [comment]
@smuzaffar  Yes, it is required in phedex and you are right. It was included in `PHEDEX-combined-web.spec` and `PHEDEX-web.spec` which @amaltaro removed. 

```
[mimran@vocms055]/% grep -irn "PHEDEX-datasvc" *
PHEDEX-combined-web.spec:4:# PHEDEX-datasvc on a combined platform of dependencies
PHEDEX-combined-web.spec:6:Requires: PHEDEX-web PHEDEX-webapp PHEDEX-datasvc
PHEDEX-datasvc.spec:1:### RPM cms PHEDEX-datasvc 2.5.0pre3
PHEDEX-datasvc.spec:41:# Getting  all PHEDEX-datasvc required sources:
PHEDEX-web.spec:16:Requires: PHEDEX-datasvc
```

**Comment by amaltaro** [comment]
Imran, perhaps we should first contact Nick/Kevin/Eric and others to see whether we still need to deploy PhEDEx in cmsweb?

**Pull request merged**